README.md

```diff
@@ -84,3 +85,66 @@ Comming soon..
 ## Usage in API design first approach
 
 Comming soon..
+
+## Building
+
+In order to save on http requests, when using one of the rulesets, the individual rules are not extended but bundled.
+In a build step all individual rules are collected and merged into the file spectral.yml.
+
+For this the cli tool [yq](https://github.com/mikefarah/yq) is used.
+So in order to build the rules you need to have it installed locally. (See [installing yq](https://github.com/mikefarah/yq#install))
+
+Also keep in mind, that the rules are built using bash scripts. So if you're on windows, you might want to use something like GitBash or [WSL](https://docs.microsoft.com/de-de/windows/wsl/about)
+
+### CI
+
+During CI the same tool and script is used to merge the Rules.
+Afterwards the script [checkSpectralYaml.sh](./util/scripts/checkSpectralYaml.sh) is executed to find out whether the commited spectral.yml is up to date.
+If that's not the case, the pipeline fails asking you to build the rules before committing.
+If [spectral.yml](./spectral.yml) is valid, the api type specific rulesets will be checked. Each of them must specify a severity for every rule inside [spectral.yml](./spectral.yml).
+
+## Testing
+
+To test our ruleset we decided to split it into individual rulesets containing only one rule.
```

Comment:
Can you provide some more motivation why separation also makes sense in regards of rule maintenance itself?